### PR TITLE
New option to automatically mark as read gone articles

### PR DESCRIPTION
**Pull request by Alkarex**
Option (off / disabled by default) to automatically and immediately mark as read entries / articles that are no longer provided in their upstream RSS / ATOM / XPath feed

### CODE REVIEW

**Comment by Alkarex** [comment]
image

image

I have slightly reordered a couple of options to have them in the same order on both the global and feed-local pages.

File: app/Models/Feed.php [hunk 1]
```diff
@@ -653,15 +653,37 @@ public function incPendingUnread(int $n = 1) {
...
 	}
 
+	/**
+	 * Remember to call updateCachedValue($id_feed) or updateCachedValues() just after.
+	 * @return int|false the number of lines affected, or false if not applicable
+	 */
 	public function keepMaxUnread() {
 		$keepMaxUnread = $this->attributes('keep_max_n_unread');
-		if ($keepMaxUnread == false) {
+		if ($keepMaxUnread === null) {
```
Comment (Alkarex) [inline app/Models/Feed.php:662]: Minor bug fixed regarding the *keep max unread* policy

**Comment by math-GH** [comment]
What could be a user story of this feature?

**Comment by Alkarex** [comment]
> What could be a user story of this feature?

* My main own usecase (for work) is to track some job position openings, by RSS or by XPath. When the jobs positions are no longer open, they disappear from the RSS feed and/or the HTML page (through XPath). With this option enabled, only the current job openings are shown in FreshRSS.

* Another usecase I have (private) is to track a Web page with a list of episodes from a TV series. When an episode is no longer available, it disappears from the HTML page (through XPath). With this option enabled, only the still-available episodes are shown in FreshRSS.

* It also allows using FreshRSS in a way similar to RSS-Bridge by re-sharing by RSS the XPath extraction of an HTML page, which stays in sync thanks to this option (I need to implement a new security token mechanism to make that truly relevant).

* It might be used to reduce information overload - one user in my family is only interested in reading the top / most recent news and gets scared away when there are too many unread articles :-P

* Finally - although this is not something I use myself - it is also relevant for an anonymous FreshRSS instance where nobody actively mark articles as read, ensuring that only current news are shown.

I hope that makes sense?

**Comment by math-GH** [comment]
Great. That is very cool and useful.

**Pull request merged**